### ISSUE TITLE
Why exception may occurs?

### ISSUE THREAD

**Issue by samarjan92** [arturbosch/detekt#160]
Gradle console output is below

21:15:39.548 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
21:15:39.548 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
21:15:39.548 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
21:15:39.548 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Where:
21:15:39.548 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Build file '/Users/d.samaryan/Documents/Adroid Samples/ktlint/Kotlindetect/app/build.gradle' line: 40
21:15:39.548 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
21:15:39.548 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
21:15:39.549 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] A problem occurred evaluating project ':app'.
21:15:39.549 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > java.lang.reflect.InvocationTargetException (no error message)
21:15:39.549 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
21:15:39.549 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
21:15:39.549 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Run with --stacktrace option to get the stack trace. 

* What went wrong:
A problem occurred evaluating project ':app'.
> java.lang.reflect.InvocationTargetException (no error message)

In my build.gradle file:
```groovy
apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply plugin: 'io.gitlab.arturbosch.detekt'


android {
    compileSdkVersion 26
    buildToolsVersion "26.0.0"
    defaultConfig {
        applicationId "com.example.kotlindetect"
        minSdkVersion 19
        targetSdkVersion 26
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}

dependencies {
    compile fileTree(dir: 'libs', include: ['*.jar'])
    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
        exclude group: 'com.android.support', module: 'support-annotations'
    })
    compile 'com.android.support:appcompat-v7:26.+'
    compile 'com.android.support.constraint:constraint-layout:1.0.2'
    testCompile 'junit:junit:4.12'
    compile "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
}
repositories {
    mavenCentral()
}
detekt {
    version = "1.0.0.M12.3"  // Specify current detekt version
    input = "$projectDir/src/main/java" // input is preconfigured to 'project.projectDir.absolutePath'
    config = "$projectDir/detekt.yml" // Use $project.projectDir to navigate inside your project
    filters = ".*test.*, .*/resources/.*" // What paths to exclude? Use comma oder semicolon to separate
}
```

Hope for your help, thanks!

**Comment by Mauin** [comment]
Do you have `$projectDir` defined somewhere? From the error message that points to line 40 I'm guessing that the variable is just not defined. Can you try replacing `$projectDir` with `$project.projectDir`?

**Comment by arturbosch** [comment]
`$projectDir` is the short form for `$project.projectDir`, learned that from @sschuberth in another ticket. He also changed the README for this.

Do you have your project on GitHub and can point us to it? Show us your `buildscript`-closure. Is this the stacktrace with `--stacktrace` enables?

**Comment by sschuberth** [comment]
> `$projectDir` is the short form for `$project.projectDir`

More generally, quoting the [project API docs](https://docs.gradle.org/4.0/userguide/writing_build_scripts.html#sec:project_api):

> Any property you access in your build script, which is not defined in the build script, is delegated to the Project object.

**Comment by samarjan92** [comment]
Thanks for the prompt reply. I prepared the gist

https://gist.github.com/samarjan92/fd1c941ffafaf5b1e420646f76ea50f9

**Comment by arturbosch** [comment]
Ok I see now. You use the new M12 version but your configuration is for pre M12. In the migration guide I wrote that M12 introduced `profiles`, so basically to fix your info wrap your `input, filters, config etc` into a `profile("main") { ... }` closure.

But I must say the error message `java.lang.IllegalArgumentException: Make sure the idea path is specified to run idea tasks!` is pretty much misleading ...

And pls use the gradle plugin `M12.4`

**Comment by samarjan92** [comment]
Thansk, I tried your advice but receive new error(

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detektGenerateConfig'.
> Could not resolve all dependencies for configuration ':detektConfig'.
   > Could not find io.gitlab.arturbosch.detekt:detekt-cli:1.0.0.M12.4.
     Searched in the following locations:
         file:/Applications/Android Studio.app/Contents/gradle/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.M12.4/detekt-cli-1.0.0.M12.4.pom
         file:/Applications/Android Studio.app/Contents/gradle/m2repository/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.M12.4/detekt-cli-1.0.0.M12.4.jar
         https://jcenter.bintray.com/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.M12.4/detekt-cli-1.0.0.M12.4.pom
         https://jcenter.bintray.com/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.M12.4/detekt-cli-1.0.0.M12.4.jar
         https://plugins.gradle.org/m2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.M12.4/detekt-cli-1.0.0.M12.4.pom
         https://plugins.gradle.org/m2/io/gitlab/arturbosch/detekt/detekt-cli/1.0.0.M12.4/detekt-cli-1.0.0.M12.4.jar
     Required by:
         project :

**Comment by arturbosch** [comment]
The detekt-gradle-plugin version must be M12.4 but deteKt itself should stay at M12.3 == when not specifying a version in the deteKt closure.

**Comment by samarjan92** [comment]
Unfortunately I keep getting errors

...
detekt {
    version = "1.0.0.M12.3"
    profile("main") {
        input = "$projectDir/src/main/java"
        config = "$projectDir/detekt.yml"
        filters = ".*test.*,.*/resources/.*,.*/tmp/.*"
    }
}
...

:app:detektCheck FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:detektCheck'.
> Could not resolve all dependencies for configuration ':app:detektCheck'.
   > Could not find io.gitlab.arturbosch.detekt:detekt-cli:1.0.0.M12.3.
     Required by:
         project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 0.871 secs

**Comment by arturbosch** [comment]
It seems like your using the old way to include gradle plugins.

You dont need this:
```
configurations {
  detekt
}
```

Try moving the `repositories`-closure before `dependencies` and add to repositories `jcenter()`

**Comment by samarjan92** [comment]
It did not help, I see same error. I push my project to repository

https://github.com/samarjan92/kotlindetect

**Comment by arturbosch** [comment]
Adding 

```
plugins {
    id "io.gitlab.arturbosch.detekt" version "1.0.0.M12.4"
}
```

to your module.app build.gradle and removing

```
buildscript {
    dependencies {
        classpath "gradle.plugin.io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.0.0.M12.4"
    }
}
```

from the project build.gradle made it work.

Android uses two gradle build files, so things can mesh up ^^

**Comment by samarjan92** [comment]
It help to avoid these erorrs, thanks, but now I receive a new error listed below:

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':detektCheck'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:243)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:236)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:225)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.process.internal.ExecException: Process 'command '/Applications/Android Studio 3.0 Preview.app/Contents/jre/jdk/Contents/Home/bin/java'' finished with non-zero exit value 255
	at org.gradle.process.internal.DefaultExecHandle$ExecResultImpl.assertNormalExitValue(DefaultExecHandle.java:380)
	at org.gradle.process.internal.DefaultJavaExecAction.execute(DefaultJavaExecAction.java:31)
	at org.gradle.api.internal.file.DefaultFileOperations.javaexec(DefaultFileOperations.java:176)
	at org.gradle.api.internal.project.DefaultProject.javaexec(DefaultProject.java:1061)
	at io.gitlab.arturbosch.detekt.DetektCheckTask.check(DetektCheckTask.kt:25)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:141)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:121)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:711)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:694)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
	... 27 more


BUILD FAILED in 1s

It happens at the end of execution gradle task, but detectGenerateConfig task have a time to generate default-detekt-config.yml file.

**Comment by samarjan92** [comment]
:detektCheck
Could not invoke null
    Reason: Can not set java.lang.String field io.gitlab.arturbosch.detekt.cli.Main.configResource to java.util.Collections$SingletonList

Usage: detekt [options]
  Options:
    --baseline, -b
      If a baseline xml file is passed in, only new code smells not in the 
      baseline are printed in the console.
    --config, -c
      Path to the config file (path/to/config.yml).
    --config-resource, -cr
      Path to the config resource on detekt's classpath (path/to/config.yml).
    --create-baseline, -cb
      Treats current analysis findings as a smell baseline for further detekt 
      runs. 
      Default: false
    --debug, -d
      Debugs given ktFile by printing its elements.
      Default: false
    --disable-default-rulesets, -dd
      Disables default rule sets.
      Default: false
    --filters, -f
      Path filters defined through regex with separator ';' (".*test.*").
    --format
      Enables formatting of source code. Cannot be used together with 
      --config. 
      Default: false
    --generate-config, -gc
      Export default config to default-detekt-config.yml.
      Default: false
    --help, -h
      Shows the usage.
    --output, -o
      Specify the file to output to.
    --output-format, -of
      Specify the output format.
      Default: XML
      Possible Values: [PLAIN, XML]
    --parallel
      Enables parallel compilation of source files. Should only be used if the 
      analyzing project has more than ~200 kotlin files.
      Default: false
  * --project, -p
      Project path to analyze (path/to/project).
    --rules, -r
      Extra paths to ruleset jars separated by ',' or ';'.
    --useTabs
      Tells the formatter that indentation with tabs are valid.
      Default: false

:detektCheck FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detektCheck'.
> Process 'command '/Applications/Android Studio 3.0 Preview.app/Contents/jre/jdk/Contents/Home/bin/java'' finished with non-zero exit value 255

**Comment by samarjan92** [comment]
I found my mistake: in the detekt clojure I set a wrong path to the project:

```groovy
detekt {
    debug = true
    version = "1.0.0.M12.3"
    profile("main") {
        input = "$projectDir/src/main/java" //   missed 'app' path element  input = "$projectDir/app/src/main/java"
        output = "$projectDir/output.xml"
        config = "$projectDir/detekt.yml"
        filters = ".*test.*,.*/resources/.*"
    }
}
```

And now plugin actually worked! Thank you so much for help in solving the problem and for your plugin!

**Comment by arturbosch** [comment]
I'm glad that it worked out! Your last exception is indeed a bug in the gradle plugin and will be addressed here #166.
